sdk/Pulumi/Deployment/Deployment_Prepare.cs

```diff
@@ -146,6 +136,173 @@ void LogExcessive(string message)
             }
         }
 
+        static async Task<T> Resolve<T>(Input<T>? input, T whenUnknown)
+        {
+            return input == null
+                ? whenUnknown
+                : await input.ToOutput().GetValueAsync(whenUnknown).ConfigureAwait(false);
+        }
+
+        private static async Task<List<Pulumirpc.Alias>> PrepareAliases(
+            Resource resource,
+            ResourceOptions options,
+            bool resourceMonitorSupportsAliasSpec)
+        {
+            var aliases = new List<Pulumirpc.Alias>();
+            if (resourceMonitorSupportsAliasSpec)
+            {
+                // map each alias in the options to its corresponding Alias proto definition.
+                foreach (var alias in options.Aliases)
+                {
+                    var resolvedAlias = await alias.ToOutput().GetValueAsync(whenUnknown: default!).ConfigureAwait(false);
+                    if (resolvedAlias == null)
+                    {
+                        // alias contains unknowns, skip it.
+                        continue;
+                    }
+
+                    var aliasSpec = new Pulumirpc.Alias.Types.Spec
+                    {
+                        Name = await Resolve(resolvedAlias.Name, ""),
+                        Type = await Resolve(resolvedAlias.Type, ""),
+                        Stack = await Resolve(resolvedAlias.Stack, Instance.StackName),
```

Comment:
These can actually default to "" as well, the engine will fill them in.

Reply:
done ✅